docs/proposals/policy.md

```diff
@@ -0,0 +1,146 @@
+
+
+Policy extends the existing authorization rules in Kubernetes to fulfill requirements around extended verb sets, verb specific attributes, role based management of attribute based access control, and more efficient evaluation of authorization status at larger scales.
+
+
+### Runtime Authorization Types
+```
+type Authorizer interface{
+	// Authorize can return both allowedBy and deniedBecause strings at the same time.  
+	// This can happen when one rule allows an action, but another denies the action.
+	// Allowing both to be returned makes it easier to track why policy is allowing or
+	// denying a particular action.  Allowed by must be non-empty and deniedBecause must
+	// be empty in order for an action to be allowed.
+	Authorize(a AuthorizationAttributes) (allowedBy string, deniedBecause string, error)
+}
+
+// AuthorizationAttributeBuilder takes a request and creates AuthorizationAttributes.  
+// Since the attributes returned can vary based on verb type, AuthorizationAttributeBuilders 
+// are paired with Authorizers during registration.
+type AuthorizationAttributeBuilder interface{
+	GetAttributes(request http.Request) (AuthorizationAttributes, error)
+}
+
+type AuthorizationAttributes interface{
+	GetUserInfo() UserInfo  // includes GetName() string GetGroups() []string
+	GetVerb() string
+	GetNamespace() string
+	// GetRequestAttributes is of type interface{} because different verbs and different 
+	// Authorizer/AuthorizationAttributeBuilder pairs may have different contract requirements
+	GetRequestAttributes() interface{}
+}
+```
+A single Authorizer/AuthorizationAttributeBuilder pair may be registered with the apiserver.
+
+
+### Verb Set
+Currently, we only have "readonly" and everything else.  This does not give sufficient control.  We need to support all of the standard HTTP verbs, plus different concepts such as "watch" that don't have equivalents.  Our authorizer will start with the following verbs, but other implementations could choose different ones:
```

Comment:
You should reference the two prior pull requests and link to their discussions in your abstract